src/librustc_trans/trans/expr.rs

```diff
@@ -133,13 +133,25 @@ pub fn trans_into<'blk, 'tcx>(bcx: Block<'blk, 'tcx>,
     ) {
         if !qualif.intersects(check_const::ConstQualif::PREFER_IN_PLACE) {
             if let SaveIn(lldest) = dest {
-                let global = consts::get_const_expr_as_global(bcx.ccx(), expr, qualif,
-                                                            bcx.fcx.param_substs);
-                // Cast pointer to destination, because constants
-                // have different types.
-                let lldest = PointerCast(bcx, lldest, val_ty(global));
-                memcpy_ty(bcx, lldest, global, expr_ty_adjusted(bcx, expr));
-                return bcx;
+                match consts::get_const_expr_as_global(bcx.ccx(), expr, qualif,
+                                                       bcx.fcx.param_substs,
+                                                       consts::TrueConst::No) {
+                    Ok(global) => {
+                        // Cast pointer to destination, because constants
+                        // have different types.
+                        let lldest = PointerCast(bcx, lldest, val_ty(global));
+                        memcpy_ty(bcx, lldest, global, expr_ty_adjusted(bcx, expr));
+                        return bcx;
+                    },
+                    Err(consts::ConstEvalFailure::Runtime(_)) => {
+                        // in case const evaluation errors, translate normally
+                        // debug assertions catch the same errors
+                        // see RFC 1229
+                    },
+                    Err(consts::ConstEvalFailure::Compiletime(_)) => {
+                        return bcx;
```

Comment:
hmm, so wait, in what scenario do we get a `Compiletime(_)` failure here?  And what is the effect of returning early from this call to `trans_into` in that case?  Does it cause the expression in question to become a no-op?

(I am essentially surprised that we are not either causing an ICE here, or folding this case into the `Runtime(_)` one immediately above.)